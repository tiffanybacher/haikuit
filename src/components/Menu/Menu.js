import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class Menu extends Component {
  render() {
    return (
      <section className="Menu">
        <nav className="Menu-sidebar">
          <button 
            className="Menu-close-btn"
            onClick={this.props.toggleMenu}>
            <i className="fas fa-times"></i>
          </button>
          <li><NavLink to="/about">About</NavLink></li>
          <li>Your Haikus</li>
        </nav>
      </section>
    );
  }
}

export default Menu;