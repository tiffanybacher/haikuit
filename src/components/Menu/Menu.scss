.Menu {
  background: rgba(black, .5);
  height: 100%;
  margin: 0;
  padding: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.Menu-sidebar {
  background: $purple;
  color: white;
  height: 100%;
  width: 60%;

  .btn-holder {
    text-align: right;
  }

  .fa-times {
    color: white;
    font-size: 1.75em;
    padding:20px;
  }

  a {
    color: white;
    text-decoration: none;
  }

  li {
    font-size: 1.25em;
    padding: 20px 10px;
    list-style-type: none;
    transition: 0.3s;
  }
}

@media only screen and (min-width: 768px) {
  .Menu-sidebar {
    width: 30%;
  }
}