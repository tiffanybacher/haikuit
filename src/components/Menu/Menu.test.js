import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  let toggleMenu;
  let wrapper;

  beforeEach(() => {
    toggleMenu = jest.fn();
    wrapper = shallow(<Menu toggleMenu={toggleMenu} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should close menu when x button is clicked', () => {
    wrapper.find('.Menu-close-btn').simulate('click');

    expect(toggleMenu).toHaveBeenCalled();
  });
});