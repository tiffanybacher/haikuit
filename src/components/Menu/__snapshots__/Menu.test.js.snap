// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot 1`] = `
<section
  className="Menu"
>
  <nav
    className="Menu-sidebar"
  >
    <div
      className="btn-holder"
    >
      <button
        className="Menu-close-btn"
        onClick={[MockFunction]}
      >
        <i
          className="fas fa-times"
        />
      </button>
    </div>
    <h1>
      <Link
        to="/"
      >
        Haikuit
        <i
          className="fas fa-feather-alt"
        />
      </Link>
    </h1>
    <ul
      className="nav-list"
    >
      <li
        onClick={[MockFunction]}
      >
        <Link
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        onClick={[MockFunction]}
      >
        <Link
          to="/about"
        >
          About
        </Link>
      </li>
      <li
        onClick={[MockFunction]}
      >
        <Link
          to="/haikus"
        >
          Your Haikus
        </Link>
      </li>
    </ul>
  </nav>
</section>
`;
